wgpu-hal/src/gles/wgl.rs

```diff
@@ -0,0 +1,750 @@
+        let extentions = extensions(&extra, dc);
+
+        let srgb_capable = extentions.contains("WGL_EXT_framebuffer_sRGB")
+            || extentions.contains("WGL_ARB_framebuffer_sRGB")
+            || gl
+                .supported_extensions()
+                .contains("GL_ARB_framebuffer_sRGB");
+
+        if srgb_capable {
+            unsafe { gl.enable(glow::FRAMEBUFFER_SRGB) };
+        }
+
+        if desc.flags.contains(InstanceFlags::VALIDATION) && gl.supports_debug() {
+            log::info!("Enabling GL debug output");
+            unsafe { gl.enable(glow::DEBUG_OUTPUT) };
+            unsafe { gl.debug_message_callback(super::gl_debug_message_callback) };
+        }
+
+        context.unmake_current().map_err(|e| {
+            crate::InstanceError::with_source(
+                String::from("unable to unset the current WGL context"),
+                e,
+            )
+        })?;
+
+        Ok(Instance {
+            inner: Arc::new(Mutex::new(Inner {
+                opengl_module,
+                gl,
+                context,
+            })),
+            srgb_capable,
+        })
+    }
+
+    #[cfg_attr(target_os = "macos", allow(unused, unused_mut, unreachable_code))]
+    unsafe fn create_surface(
+        &self,
+        _display_handle: RawDisplayHandle,
+        window_handle: RawWindowHandle,
+    ) -> Result<Surface, crate::InstanceError> {
+        let window = if let RawWindowHandle::Win32(handle) = window_handle {
+            handle
+        } else {
+            return Err(crate::InstanceError::new(format!(
+                "unsupported window: {window_handle:?}"
+            )));
+        };
+        let dc = unsafe { GetDC(window.hwnd as *mut _) };
+        if dc.is_null() {
+            return Err(crate::InstanceError::with_source(
+                String::from("unable to get the device context from window"),
+                Error::last_os_error(),
+            ));
+        }
+        Ok(Surface {
+            dc,
+            presentable: true,
+            swapchain: None,
+            srgb_capable: self.srgb_capable,
+        })
+    }
+    unsafe fn destroy_surface(&self, _surface: Surface) {}
+
+    unsafe fn enumerate_adapters(&self) -> Vec<crate::ExposedAdapter<super::Api>> {
+        unsafe {
+            super::Adapter::expose(AdapterContext {
+                inner: self.inner.clone(),
+            })
+        }
+        .into_iter()
+        .collect()
+    }
+}
+
+pub struct Swapchain {
+    surface_context: WglContext,
+    surface_gl: glow::Context,
+    framebuffer: glow::Framebuffer,
+    renderbuffer: glow::Renderbuffer,
+    /// Extent because the window lies
+    extent: wgt::Extent3d,
+    format: wgt::TextureFormat,
+    format_desc: super::TextureFormatDesc,
+    #[allow(unused)]
+    sample_type: wgt::TextureSampleType,
+}
+
+pub struct Surface {
+    // FIXME: Find a way to call ReleaseDC on the same thread as GetDC.
```

Comment:
Maybe when we drop the surface, clear it if it's being dropped from the same thread, otherwise log.

Reply:
I found a way to use it on-demand on the same thread.